src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaTransportControls.MediaPlayer.cs

```diff
@@ -0,0 +1,239 @@
+
+namespace Windows.UI.Xaml.Controls
+{
+	public partial class MediaTransportControls : Control
+	{
+		private Windows.Media.Playback.MediaPlayer _mediaPlayer;
+		private bool _isScrubbing = false;
+
+		internal void SetMediaPlayer(Windows.Media.Playback.MediaPlayer mediaPlayer)
+		{
+			UnbindMediaPlayer();
+
+			_mediaPlayer = mediaPlayer;
+
+			if (_playPauseButton != null)
+			{
+				BindMediaPlayer();
+			}
+		}
+
+		private void BindMediaPlayer()
+		{
+			_mediaPlayer.PlaybackSession.PlaybackStateChanged += OnPlaybackStateChanged;
+			_mediaPlayer.PlaybackSession.BufferingProgressChanged += OnBufferingProgressChanged;
+			_mediaPlayer.PlaybackSession.NaturalDurationChanged += OnNaturalDurationChanged;
+			_mediaPlayer.PlaybackSession.PositionChanged += OnPositionChanged;
+
+			_playPauseButton.Click += PlayPause;
+			_playPauseButtonOnLeft.Click += PlayPause;
+			_audioMuteButton.Click += ToggleMute;
+			_volumeSlider.ValueChanged += OnVolumeChanged;
+			_stopButton.Click += Stop;
+			_skipForwardButton.Click += SkipForward;
+			_skipBackwardButton.Click += SkipBackward;
+
+			AttachThumbEventHandlers(_progressSlider);
+		}
+
+		private void UnbindMediaPlayer()
+		{
+			try
+			{
+				_mediaPlayer.PlaybackSession.PlaybackStateChanged -= OnPlaybackStateChanged;
+				_mediaPlayer.PlaybackSession.BufferingProgressChanged -= OnBufferingProgressChanged;
+				_mediaPlayer.PlaybackSession.NaturalDurationChanged -= OnNaturalDurationChanged;
+				_mediaPlayer.PlaybackSession.PositionChanged -= OnPositionChanged;
+
+				_playPauseButton.Click -= PlayPause;
+				_playPauseButtonOnLeft.Click -= PlayPause;
+				_audioMuteButton.Click -= ToggleMute;
+				_volumeSlider.ValueChanged -= OnVolumeChanged;
+				_stopButton.Click -= Stop;
+				_skipForwardButton.Click -= SkipForward;
+				_skipBackwardButton.Click -= SkipBackward;
+
+				DetachThumbEventHandlers(_progressSlider);
+			}
+			catch (Exception ex)
+			{
+				this.Log().ErrorIfEnabled(() => $"Unable to unbing MediaTransportControls properly: {ex.Message}", ex);
+			}
+		}
+
+		private void OnPlaybackStateChanged(MediaPlaybackSession sender, object args)
+		{
+			var state = (MediaPlaybackState)args;
+			
+			switch (state)
+			{
+				case MediaPlaybackState.Opening:
+				case MediaPlaybackState.Paused:
+				case MediaPlaybackState.None:
+					CancelControlsVisibilityTimer();
+					VisualStateManager.GoToState(this, "PlayState", false);
+					VisualStateManager.GoToState(this, "Normal", false);
+					break;
+				case MediaPlaybackState.Playing:
+					ResetControlsVisibilityTimer();
+					VisualStateManager.GoToState(this, "PauseState", false);
+					VisualStateManager.GoToState(this, "Normal", false);
+					break;
+				case MediaPlaybackState.Buffering:
+					VisualStateManager.GoToState(this, "Buffering", false);
+					break;
+			}
+		}
+
+		private void OnBufferingProgressChanged(MediaPlaybackSession sender, object args)
+		{
+
+			Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
```

Comment:
why Dispatch? (and below)

Reply:
Dispatched because it's UI update made from player background thread